tests/std/tests/P2404R3_move_only_types_for_comparison_concepts/test.compile.pass.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <compare>
+#include <concepts>
+#include <memory>
+
+using namespace std;
+
+struct MoveOnly {
+    MoveOnly(int); // not defined
+    MoveOnly(const MoveOnly&)            = delete;
+    MoveOnly& operator=(const MoveOnly&) = delete;
+    MoveOnly(MoveOnly&&)                 = default;
+    MoveOnly& operator=(MoveOnly&&)      = default;
+
+    strong_ordering operator<=>(const MoveOnly&) const = default;
+    strong_ordering operator<=>(int) const; // not defined
+};
+
+// Check three_way_comparable_with
+static_assert(three_way_comparable_with<MoveOnly, int>);
+static_assert(three_way_comparable_with<int, MoveOnly>);
+static_assert(three_way_comparable_with<MoveOnly, int, strong_ordering>);
+static_assert(three_way_comparable_with<int, MoveOnly, strong_ordering>);
+
+// Check equality_comparable_with
+static_assert(equality_comparable_with<MoveOnly, int>);
+static_assert(equality_comparable_with<int, MoveOnly>);
+static_assert(equality_comparable_with<unique_ptr<int>, nullptr_t>);
+static_assert(equality_comparable_with<nullptr_t, unique_ptr<int>>);
+
+// Check totally_ordered_with
+static_assert(totally_ordered_with<MoveOnly, int>);
+static_assert(totally_ordered_with<int, MoveOnly>);
+
+// Check [diff.cpp20.concepts]
+// clang-format off
+template <class T, class U>
+    requires equality_comparable_with<T, U>
+bool attempted_equals(const T&, const U& u); // not defined

```

Comment:
Hyper nitpick: This was present in WG21-P2404R3, where `const T&` here is unnamed, but `u` is named, and below both `t` and `u` are named.

Since these functions aren't defined, and we care only about the parameter types and have no reason to think about the parameters themselves, we should just drop the names.